python/lsst/skymap/packers.py

```diff
@@ -31,10 +69,46 @@ class SkyMapDimensionPacker(DimensionPacker):
     Parameters
     ----------
     fixed : `lsst.daf.butler.DataCoordinate`
-        Expanded data ID that must include at least the skymap dimension.
-    dimensions : `lsst.daf.butler.DimensionGraph`
+        Data ID that identifies just the ``skymap`` dimension.  Must have
+        dimension records attached unless ``n_tracts`` and ``n_patches`` are
+        not `None`.
+    dimensions : `lsst.daf.butler.DimensionGraph`, optional
         The dimensions of data IDs packed by this instance.  Must include
-        skymap, tract, and patch, and may include band.
+        ``{skymap, tract, patch}``, and may include ``band``.  If not provided,
+        this will be set to include ``band`` if if ``n_bands != 0``.
+    bands : `~collections.abc.Mapping` [ `str`, `int` ] or `None`, optional
+        Mapping from band name to integer to use in the packed ID.  `None` uses
+        a fixed set of bands defined in this class, but this should be avoided
+        in new code.
+    n_bands : `int` or `None`, optional
+        The number of bands to leave room for in the packed ID.  If `None`,
```

Comment:
Maybe worth mentioning here why one might want `n_bands != len(bands)`? `>` makes sense to me, to leave room for future changes, but should `<` be a logic or value error? I didn't make the connection as to why both of these could be passed until I looked at the logic of the code below.

Reply:
I could see `<` being useful especially when it comes from the config interface, as a way to more easily have single consistent mapping of which sometimes only a subset is used (e.g. broadband + narrowband filters).

Reply:
I think that's worth a sentence in this docstring, because I don't feel it's an obvious usage.